### PR TITLE
Add CONTRIBUTING.adoc

### PR DESCRIPTION
**Pull request by sofezticated-gentleman**

### CODE REVIEW

**Comment by sofezticated-gentleman** [comment]
@HelioGuilherme66 , I have creted a `CONTRIBUTIONS.adoc` file if there are any issues or changes you would like to see please tell me

File: CONTRIBUTIONS.adoc [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+we try to write
+http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html[idiomatic
+Python] and follow the
+https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)[SOLID
+principles] with all new code. An important guideline is that the code
+should be clear enough that comments are generally not needed.
+
+All code, including test code, must be compatible with all supported
+Python interpreters and versions. Most importantly this means that the
+code must support both Python 2 and Python 3.
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:129]: >must support both Python 2 and Python 3.
must support Python 3 (actually, greater than 3.6). RIDE's code still have some Python 2 conditioning, due to its dual compatibility on version 1.7.4.2, and also in the included Robot Framework library from version 3.1.2.

File: CONTRIBUTIONS.adoc [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+==== Line length
+
+Maximum line length with Python code, including docstrings and comments,
+is 88 characters. This is also what
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:134]: >is 88 characters. This is also what

is 100 characters. The recommended size of 88 characters, is what

File: CONTRIBUTIONS.adoc [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+https://pypi.org/project/black/[Black] uses by default and
+https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length[their
+documentation] explains why. Notice that we do not have immediate plans
+to actually take Black into use but we may consider that later.
+
+With Robot Framework tests the maximum line length is 100.
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:140]: >With Robot Framework tests the maximum line length is 100.

[Remove this line]

File: CONTRIBUTIONS.adoc [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+https://www.python.org/dev/peps/pep-0257/[PEP-257]. See
+link:#api-documentation[API documentation] section below for more
+details about documentation syntax, generating API docs, etc.
+
+=== Documentation
+
+With new features adequate documentation is as important as the actual
+functionality. Different documentation is needed depending on the issue.
+
+==== User Guide
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:172]: >==== User Guide

==== Robot Framework's User Guide

File: CONTRIBUTIONS.adoc [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+Robot Framework's features are explained in the
+http://robotframework.org/robotframework/#user-guide[User Guide]. It is
+generated using a custom script based on the source in
+http://docutils.sourceforge.net/rst.html[reStructuredText] format. For
+more details about editing and generating it see
+doc/userguide/README.rst.
+
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:180]: === RIDE's Wiki

https://github.com/robotframework/RIDE/wiki[RIDE's Wiki] is the preferred location for the documentation. This is to promote collaboration from users, and hopefully, to have the content easily updated.

File: CONTRIBUTIONS.adoc [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+==== Libraries
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:181]: This section must be properly adjusted to RIDE:
- we don't provide Libraries
- important to know how to run unit tests
- important to know how to run RIDE (with invoke)
- important to know how to run RIDE using the debug option

File: CONTRIBUTIONS.adoc [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+These guidelines instruct how to submit issues and contribute code or
+documentation to the
+https://github.com/robotframework/RIDE[RIDE]. There are also many other projects in the larger
+http://robotframework.org[Robot Framework ecosystem] that you can
+contribute to. If you notice a library or tool missing, there is hardly
+any better way to contribute than creating your own project. Other great
+ways to contribute include answering questions and participating
+discussion on
+https://groups.google.com/forum/#!forum/robotframework-users[robotframework-users]
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:11]: Add the preferred Forum (like in README: https://forum.robotframework.org/c/tools/ride/21)

File: CONTRIBUTIONS.adoc [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+source as well as why and how to contribute to open source projects. If
+you are totally new to these topics, it may be a good idea to look at
+the generic https://opensource.guide/[Open Source Guides] first.
+
+== Submitting issues
+
+Bugs and enhancements are tracked in the
+https://github.com/robotframework/RIDE/issues[issue tracker].
+If you are unsure if something is a bug or is a feature worth
+implementing, you can first ask on link:[robotframework-users] mailing
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:25]: (again add the Forum as first medium)

File: CONTRIBUTIONS.adoc [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+list, on
+http://webchat.freenode.net/?channels=robotframework&prompt=1[IRC]
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:27]: This IRC is rarely used, so it is better to remove.

File: CONTRIBUTIONS.adoc [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+(#robotframework on irc.freenode.net), or on
+https://robotframework-slack-invite.herokuapp.com[Slack]. These and
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:29]: Important to mention the **#ride** channel.

File: CONTRIBUTIONS.adoc [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+comments to the existing issue instead of creating a new one.
+
+=== Reporting bugs
+
+
+Explain the bug you have encountered so that others can understand it
+and preferably also reproduce it. Key things to have in good bug report:
+
+. Version information
+* Robot Framework version
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:44]: * RIDE version

File: CONTRIBUTIONS.adoc [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+and we do not go through them in more detail. We do, however, recommend
+to create dedicated topic branches for pull requests instead of creating
+them based on the master branch. This is especially important if you
+plan to work on multiple pull requests at the same time.
+
+=== Coding conventions
+
+==== General guidelines
+
+Robot Framework uses the general Python code conventions defined in
```
Comment (HelioGuilherme66) [inline CONTRIBUTIONS.adoc:118]: RIDE

or

Robot Framework and RIDE

**Review by HelioGuilherme66 — CHANGES_REQUESTED** [review]
Overall this is what we needed. Good job!

We need to remove specific parts of Robot Framework, and replace with RIDE ones.

You should add a special reference to wxPython, near the coding section. For example:

RIDE graphical user interface, can only exist due the use of wxPython (from wxWidgets). Users wanting to write Python code for RIDE, should be acquainted with https://wxpython.org[wxPython] in particular by exploring its demo.

**Comment by HelioGuilherme66** [comment]
Please rename the file to `CONTRIBUTING.adoc` (this is the standard name).

**Comment by sofezticated-gentleman** [comment]
okay @HelioGuilherme66  I will do the required changes and make a pr by today itself

**Comment by HelioGuilherme66** [comment]
> will do the required changes and make a pr by today itself

@sofezticated-gentleman No need for PR, you just have to make the changes, commit and push to your branch. This PR will be auto-updated.

**Comment by sofezticated-gentleman** [comment]
@HelioGuilherme66 is this link still relevant : https://www.tutorialspoint.com/robot_framework/robot_framework_first_test_case_using_ride.htm and this too : https://www.softwaretestinghelp.com/getting-started-with-robot-framework-ride/

**Comment by sofezticated-gentleman** [comment]
> Please rename the file to `CONTRIBUTING.adoc` (this is the standard name).

okay

**Comment by sofezticated-gentleman** [comment]
@HelioGuilherme66 @JFoederer please review the pr

**Review by HelioGuilherme66 — APPROVED** [review]
Will improve the content by using links to other documents, in future releases.

**Comment by sofezticated-gentleman** [comment]
Hey @HelioGuilherme66 can you please accept it as a hacktober fest issue?

**Comment by sofezticated-gentleman** [comment]
@HelioGuilherme66 are there any issues or ways i can contribute! i have basic ml knowledge and im comfortable python and its libs?

**Comment by HelioGuilherme66** [comment]
@sofezticated-gentleman Are you able to use Slack? We should not be communicating here.

**Pull request merged**